src/controllers/client.js

```diff
@@ -115,24 +115,36 @@ const updateDocWithUPI = async (echisClientDoc, clientNumber) => {
       logger.error(GET_ECHIS_DOC_FAILED);
     });
 };
-const updateDocWithUpdateError = async (echisClientDoc) => {
-  getEchisDocForUpdate(echisClientDoc.doc._id)
-    .then((echisDoc) => {
-      updateEchisDocWithFailedIdentification(echisDoc)
-        .then((echisResponse) => {
-          logger.information(
-            `${CLIENT_UPDATE} ${JSON.stringify(echisResponse)}`
-          );
-          return echisResponse;
-        })
-        .catch((error) => {
-          logger.error(UPDATE_ECHIS_DOC_FAILED);
-          logger.error(error);
-        });
-    })
-    .catch((error) => {
-      logger.error(GET_ECHIS_DOC_FAILED);
+const createClientDetailsMismatchReport = async (echisClientDoc, fields) => {
+  try {
+    logger.information("Creating details mismatch report in echis");
+    const axiosInstance = axios.create({
+      baseURL: CHT.url,
+      headers: {
+        "Content-Type": "application/json",
+      },
+      auth: {
+        username: CHT.username,
+        password: CHT.password,
+      },
     });
+    const body = {
+      _meta: {
+        form: "client_details_mismatch",
+      },
+      patient_id: echisClientDoc.doc._id,
+      authored_on: echisClientDoc.doc.reported_date,
+      mismatched_fields: fields,
+      source: `Client Registry`,
+    };
+    const response = await axiosInstance.post(`api/v2/records`, body);
+    logger.information(COMPLETED_SUCCESSFULLY);
+    return response;
+
+  } catch (error) {
+    logger.error(error);
+    return error;
+  }
 };
```

Comment:
Still a lot ongoing here. We can retain it for now, but this is a good candidate for refactoring.